### ISSUE TITLE
Implementing unique file names  

### ISSUE THREAD

**Issue by eugenemiretsky** [dwilkie/carrierwave_direct#42]
Hi,
How would one go about implementing unique file names. 
In carrierwave one can just define 
def filename
   # generate unique  file name 
end
However, with carrierwave direct the file is being uploaded directly from the front end.

Any ideas?

**Comment by matthewhatch** [comment]
I know this issue is over 3 years old, but I'm running into the same problem, the unique GUID is the same for uploads from SPA.  This causes an issue when a user has a file called logo.png and another also called logo.png (a different file, locally `/dirA/logo.png` and `/dirB/logo.png'

**Comment by matthewhatch** [comment]
So in my case the reason the GUID in the file path wasn't changing was due to the way I was generating the instance of the uploader class in javascript.  I'm using gon to access rails variable from javascript and was using it to access my uploader.  The uploader was generated when I hit the home controller and that was it.  Every time I accessed the uploader it was the same instance of the uploader class.  I was able to get around this by using `gon.watch` to create a new instance of the class.
